index.js

```diff
@@ -97,84 +97,167 @@ const parseArgs = (
-        // isn't used ie. both --foo=b and --foo b should work
-
-        // If withValue option is specified, take next position argument as
-        // value and then increment pos so that we don't re-evaluate that
-        // arg, else set value as undefined ie. --foo b --bar c, after setting
-        // b as the value for foo, evaluate --bar next and skip 'b'
-        const val = options.withValue &&
-          ArrayPrototypeIncludes(options.withValue, arg) ? argv[++pos] :
-          undefined;
-        storeOptionValue(options, arg, val, result);
+        optionKey = StringPrototypeSlice(arg, 2, index);
+        optionValue = StringPrototypeSlice(arg, index + 1);
       } else {
-        // Cases when an arg is specified without a value, example
-        // '--foo --bar' <- 'foo' and 'bar' flags should be set to true and
-        // save value as undefined
-        storeOptionValue(options, arg, undefined, result);
+        // e.g. '--foo'
+        optionKey = StringPrototypeSlice(arg, 2);
+        if (isExpectingValue(optionKey, options) && isOptionValue(nextArg)) {
+          // e.g. '--foo' 'bar'
+          optionValue = nextArg;
+          pos++;
+        }
       }
-
-    } else {
-      // Arguments without a dash prefix are considered "positional"
-      ArrayPrototypePush(result.positionals, arg);
+      storeOptionValue(optionKey, optionValue, options, result);
+      pos++;
+      continue;
     }
 
+    // Anything that did not get handled above is a positional.
+    ArrayPrototypePush(result.positionals, arg);
     pos++;
   }
 
   return result;
 };
 
+/**
+  * Determines if `arg` is a just a short option.
+  * @example '-f'
+  */
+function isLoneShortOption(arg) {
+  return arg.length === 2 &&
+    StringPrototypeCharAt(arg, 0) === '-' &&
+    StringPrototypeCharAt(arg, 1) !== '-';
+}
+
+/**
+  * Determines if `arg` is a short option group.
+  *
+  * See Guideline 5 of the [Open Group Utility Conventions](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html).
+  *   One or more options without option-arguments, followed by at most one
+  *   option that takes an option-argument, should be accepted when grouped
+  *   behind one '-' delimiter.
+  * @example
+  * isShortOptionGroup('-a', {}) // returns false
+  * isShortOptionGroup('-ab', {}) // returns true
+  * isShortOptionGroup('-fb', { withValue: ['f'] }) // returns false
+  * isShortOptionGroup('-bf', { withValue: ['f'] }) // returns true
+  */
+function isShortOptionGroup(arg, options) {
+  if (arg.length <= 2) return false;
+  if (StringPrototypeCharAt(arg, 0) !== '-') return false;
+  if (StringPrototypeCharAt(arg, 1) === '-') return false;
+
+  const onlyFlags = arg.slice(1, -1);
+  for (let index = 0; index < onlyFlags.length; index++) {
+    const optionKey = getOptionKey(StringPrototypeCharAt(onlyFlags, index));
+    if (isExpectingValue(optionKey, options)) {
+      return false;
+    }
+  }
+  return true;
```

Comment:
```suggestion
  return ArrayPrototypeSome(onlyFlags, (char) => {
    const optionKey = getOptionKey(char);
    return !isExpectingValue(optionKey, options);
  });
```

Reply:
Got rid of the explicit loop and early return.